src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -508,11 +508,7 @@ internal SynthesizedAttributeData SynthesizeTupleNamesAttributeOpt(TypeSymbol ty
             Debug.Assert((object)stringType != null);
             var names = TupleNamesEncoder.Encode(type, stringType);
 
-            // If there are no names, elide the attribute entirely
-            if (names.IsDefault)
-            {
-                return null;
-            }
+            Debug.Assert(!names.IsDefault, "should not need the attribute when no tuple names");
```

Comment:
It looks like the name is still `SynthesizeTupleNamesAttributeOpt` but it should probably just be `SynthesizeTupleNamesAttribute` if the method can no longer return `null` (which I think is the case now).